test/context_test.rb

```diff
@@ -222,17 +222,17 @@ def test_log_unfinished_spans
       case i
       when 0
         assert_match(
-          /D,.*DEBUG -- ddtrace: \[ddtrace\].*\) root span parent closed but has 2 unfinished spans:/,
+          /D,.*DEBUG -- \[ddtrace\].*\) root span parent closed but has 2 unfinished spans:/,
```

Comment:
@delner, I have no idea why I needed to make this change. I don't see where in `Datadog::Logger` I made a change that would result in a different log message being emitted?

There are other places that fail for the same reason as this, but I am hesitant to change these because it might indicate I changed something in an unexpected way.

Reply:
But it also only fails on older versions of Ruby?

Rather confused here.

Reply:
Looks like there is some difference between how ruby pre 2.1 and 2.1+ handles logs.

Wonder if it might be related to this case:
https://github.com/DataDog/dd-trace-rb/blob/master/test/monkey_test.rb#L23

Without digging more into difference between outputs between versions is hard for me to understand the source of this discrepancy :|

Reply:
Ah, that `lines.each` is because `StringIO.string.lines` isn't an array in older versions of Ruby.

Reply:
I mentioned that because of the `\n` at the end of assertion error `Expected /W,.*WARN -- ddtrace: \[ddtrace\] Monkey#autopatch_modules/ to match "W, [2019-05-16T16:36:55.577974 #1871]  WARN -- [ddtrace] Monkey#autopatch_modules:\n".`

Now looking 2nd time at this string, I completely missed where the difference actually is.

Reply:
@brettlangdon 
Looks to me like the whole format got shifted to the left a bit. Where what used to be progname is now a message. 

From the logger formatting docs it should follow this format:
`SeverityID, [DateTime #pid] SeverityLabel -- ProgName: message`
instead it looks like 
`SeverityID, [DateTime #pid] SeverityLabel -- message:` which also explains the `:` at the end which I was wondering about.

Reply:
👍